Microsoft.Identity.Web/ClaimsPrincipalExtensions.cs

```diff
@@ -21,10 +21,18 @@ public static class ClaimsPrincipalExtensions
         public static string GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
         {
             string userObjectId = claimsPrincipal.GetObjectId();
+            string nameIdentifierId = claimsPrincipal.GetNameIdentifierId();
             string tenantId = claimsPrincipal.GetTenantId();
+            string policyId = claimsPrincipal.GetPolicyId();
 
-            if (!string.IsNullOrWhiteSpace(userObjectId) && !string.IsNullOrWhiteSpace(tenantId))
+            if (!string.IsNullOrWhiteSpace(nameIdentifierId) && !string.IsNullOrWhiteSpace(tenantId) && !string.IsNullOrWhiteSpace(policyId))
```

Comment:
This is actually a discussion that @kalyankrishna1 and @jmprieur had about using `sub` instead of `oid`. We have to decide if we will use `sub` on AAD cases as well.